client/app/core/authentication-api.factory.js

```diff
@@ -10,13 +10,51 @@ function base64encode (str, encoding = 'utf-8') {
 /** @ngInject */
 export function AuthenticationApiFactory ($http, API_BASE, Session, Notifications) {
   var service = {
+    globalLogin: globalLogin,
     login: login,
+    oidcLogin: oidcLogin
   };
 
   return service;
 
+  function globalLogin (authMode, userLogin, password, access_token) {
+    if (authMode == 'oidc') {
+      self.currentAuthMode = authMode
+      return oidcLogin(access_token)
+    } else {
+      self.currentAuthMode = 'database'
+      return login(userLogin, password)
+    }
+  }
+
+  function oidcLogin (access_token) {
+    return new Promise((resolve, reject) => {
+
+      return $http.get(API_BASE + '/api/auth?requester_type=ui', {
+        headers: {
+          'Authorization': 'Bearer ' + access_token,
+          'X-Auth-Token': undefined,
+          'X-Requested-With': 'XMLHttpRequest',
+        },
+      }).then(loginSuccess, loginFailure);
+
+      function loginSuccess (response) {
+        Session.setAuthToken(response.data.auth_token)
+        Session.setAuthMode(self.currentAuthMode)
+        resolve(response)
+      }
+
+      function loginFailure (response) {
```

Comment:
you don't need the extra copies of `loginSuccess` and `loginFailure` if they're the same